processing/src/main/java/io/druid/query/timeseries/DefaultTimeseriesQueryMetrics.java

```diff
@@ -35,11 +35,18 @@ public DefaultTimeseriesQueryMetrics(ObjectMapper jsonMapper)
   public void query(TimeseriesQuery query)
   {
     super.query(query);
+    limit(query);
     numMetrics(query);
     numComplexMetrics(query);
     granularity(query);
   }
 
+  @Override
+  public void limit(TimeseriesQuery query)
+  {
+    setDimension("limit", toString().valueOf(query.getLimit()));
```

Comment:
`toString().valueOf(query.getLimit())` is just a weird way of writing `String.valueOf(query.getLimit())`, but using a strange Java feature that lets you call static methods (like valueOf) as if they were methods of specific objects. So please replace it with the latter.

Reply:
this is pretty weird, now that i look at it, not sure what was i thinking.